hazelcast/src/main/java/com/hazelcast/spi/SchedulingExecutorService.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * {@link ExecutorService} which can schedule a command to run after a given delay or to execute it periodically.
+ *
+ * The {@link #scheduleWithRepetition(Runnable, long, long, TimeUnit)} has similar semantic
+ * to {@link java.util.concurrent.ScheduledExecutorService#scheduleAtFixedRate(Runnable, long, long, TimeUnit)}. It
+ * guarantees a task won't be executed by multiple threads concurrently. The difference is that this service will simply
+ * skip a scheduled execution if another thread is still running the same task instead of postponing its execution.
+ * To emphasize this difference the method is called <code>scheduleWithRepetition</code>
+ * instead of <code>scheduleAtFixedRate</code>
+ *
+ * The other difference is this service does not offer an equivalent of
+ * {@link java.util.concurrent.ScheduledExecutorService#scheduleWithFixedDelay(Runnable, long, long, TimeUnit)}
+ *
+ */
+public interface SchedulingExecutorService extends ExecutorService {
```

Comment:
I had to read multiple time to see the difference at all. So yes; please fix it. Else you also introduce a nasty class very everyone using Hazelcast since they will get this name in their popup.

I would go for 'Scheduler'